workspaces/multi-source-security-viewer/plugins/multi-source-security-viewer/src/hooks/useDarkTheme.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { useTheme } from '@material-ui/core/styles';
+
+const THEME_DARK = 'dark';
+const THEME_DARK_CLASS = 'pf-v6-theme-dark';
+
+export const useDarkTheme = () => {
```

Comment:
You should not need to do anything like this as it's all handled by MUI. 😕

Reply:
@awanlin Just adding a bit more context, we use a component library called [@aonic-ui/pipelines](https://github.com/redhat-developer/aonic-ui/tree/main/packages/pipelines/src) to re-use some react components which is built on top of Patternfly UI library, so this hook handles the dark theme for patternfly styles.

Reply:
Yeah, it's just this is all provided by MUI and Backstage. It's not clear to me why you need to bring in a different library.

Reply:
The pipelines component uses patternfly and not MUI yet. To be able to have this in dark mode, when dark mode is enabled. We have to use this hook to set the proper mode based on the current MUI theme mode.